kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ReplaceQuestions.vue

```diff
@@ -1,35 +1,347 @@
+        <KGridItem
+          style="text-align: left;"
+          :layout12="{ span: 8 }"
+          :layout8="{ span: 6 }"
+          :layout4="{ span: 3 }"
+        >
+          {{ numberOfSelectedReplacements$({
+            count: replacements.length,
+            total: selectedActiveQuestions.length }
+          )
+          }}
+        </KGridItem>
+        <KGridItem
+          style="text-align: right;"
+          :layout12="{ span: 4 }"
+          :layout8="{ span: 2 }"
+          :layout4="{ span: 1 }"
+        >
+          <KButton
+            :primary="true"
+            :text="replaceAction$()"
+            :disable="replacements.length !== selectedActiveQuestions.length"
+            @click="handleReplacement"
+          />
+        </KGridItem>
+      </KGrid>
+    </div>
+    <KModal
+      v-if="showCloseConfirmation"
+      :submitText="coreString('continueAction')"
+      :cancelText="coreString('cancelAction')"
+      :title="closeConfirmationTitle$()"
+      @cancel="handleCancelClose"
+      @submit="handleConfirmClose"
+    >
+      {{ closeConfirmationMessage$() }}
+    </KModal>
   </div>
 
 </template>
 
 
 <script>
 
+  import { enhancedQuizManagementStrings } from 'kolibri-common/strings/enhancedQuizManagementStrings';
+  import { getCurrentInstance, computed, ref } from 'kolibri.lib.vueCompositionApi';
+  import { get } from '@vueuse/core';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
   import { injectQuizCreation } from '../../../composables/useQuizCreation';
+  import { PageNames } from '../../../constants/index';
+  import AccordionItem from './AccordionItem';
+  import AccordionContainer from './AccordionContainer';
 
   export default {
     name: 'ReplaceQuestions',
-    setup() {
+    components: {
+      AccordionContainer,
+      AccordionItem,
+    },
+    mixins: [commonCoreStrings],
+    setup(_, context) {
+      const router = getCurrentInstance().proxy.$router;
+      const {
+        replaceQuestions$,
+        deleteSectionLabel$,
+        replaceAction$,
+        selectAllLabel$,
+        closeConfirmationMessage$,
+        closeConfirmationTitle$,
+        replaceQuestionsExplaination$,
+        numberOfSelectedReplacements$,
+        numberOfQuestionsReplaced$,
+      } = enhancedQuizManagementStrings;
       const {
         // Computed
+        activeQuestions,
+        activeSection,
         selectedActiveQuestions,
+        activeResourceMap,
+        replacementQuestionPool,
+        clearSelectedQuestions,
+        toggleItemState,
+        isItemExpanded,
+        selectAllReplacementQuestions,
+        replaceSelectedQuestions,
+        toggleQuestionInSelection,
+        updateSection,
       } = injectQuizCreation();
+
+      const showCloseConfirmation = ref(false);
+
+      function handleCancelClose() {
+        showCloseConfirmation.value = false;
+      }
+
+      function handleConfirmClose() {
+        context.emit('closePanel');
+      }
```

Comment:
This ends up being repeated in each of the child components, once for each KModal and I wish I could think of a way to clean this up somehow. That or maybe if this is common to every use of KModal we could simplify this w/ a module in KDS that could make defining KModal's "show/hide/@confirm/@cancel" in a nicer way.